java/code/src/com/suse/manager/clusters/ClusterManager.java

```diff
@@ -0,0 +1,609 @@
+                                itemVal.putAll(nestedVal);
+                            }
+
+                        });
+                        items.add(itemVal);
+                    }
+                }
+                else {
+                    throw new RuntimeException("$value for edit-group doesn't evaluate to a collection");
+                }
+            }
+            else {
+                map.forEach((key, valExpr) -> {
+                    if (valExpr instanceof String) {
+                        Object val = evalExpression((String) valExpr, context);
+                        out.put(key, val);
+                    }
+                    else if (valExpr instanceof Map) {
+                        Map nestedVal = new HashMap();
+                        evalFormInitialData(nestedVal, key, (Map<String, Object>)valExpr, context);
+                        out.putAll(nestedVal);
+                    }
+                });
+            }
+    }
+
+    /**
+     * Get state hooks for cluster provider.
+     * @param provider cluster provider label
+     * @return state hooks as Map (if any)
+     */
+    public Optional<Map<String, List<String>>> getStateHooks(String provider) {
+        return FormulaFactory.getClusterProviderMetadata(provider, "state_hooks", Map.class)
+                .map(m -> (Map<String, List<String>>)m);
+
+    }
+
+    /**
+     * Get the details fields to show for each cluster node.
+     * @param provider cluster provider label
+     * @return detail fields to show (if any)
+     */
+    public Optional<List<String>> getNodesListFields(String provider) {
+        return FormulaFactory.getClusterProviderMetadata(provider, "ui:nodes_list:fields", List.class)
+                .map(l -> (List<String>)l);
+    }
+
+    /**
+     * Update cluster name and description.
+     * @param cluster cluster
+     * @param name new name
+     * @param description new description
+     */
+    public void update(Cluster cluster, String name, String description) {
+        cluster.setName(name);
+        cluster.setDescription(description);
+        getSession().save(cluster);
+        cluster.getGroup().setName(getGroupName(name));
+        cluster.getGroup().setDescription(getGroupDescription(name));
+        ServerGroupFactory.save(cluster.getGroup());
+    }
+
+    private String getGroupName(String clusterName) {
+        return "Cluster " + clusterName;
+    }
+
+    private String getGroupDescription(String clusterName) {
+        return "Group for cluster " + clusterName;
+    }
+
+    /**
+     * Find a cluster by id and user. Checks if the cluster is in the same org as the user.
+     * @param id cluster ir
+     * @param user user
+     * @return the cluster if found
+     */
+    public Optional<Cluster> findClusterByIdAndUser(Long id, User user) {
+        return ClusterFactory.findClusterById(id)
+                .filter(c -> c.getOrg().equals(user.getOrg()));
+    }
```

Comment:
It's better the use `ClusterFactory.findClusterByIdAndOrg()` instead.

Reply:
Fixed